core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -507,17 +542,21 @@ public void stopPartitions(Set<StopPartition> stopPartitions,
                 LOGGER.error("Error while stopping the partition: {}", stopPartition, ex);
             }
         }
-        // Note `deleteLocalLog` will always be true when `deleteRemoteLog` is true but not the other way around.
         Set<TopicIdPartition> deleteLocalPartitions = stopPartitions.stream()
                 .filter(sp -> sp.deleteLocalLog() && topicIdByPartitionMap.containsKey(sp.topicPartition()))
                 .map(sp -> new TopicIdPartition(topicIdByPartitionMap.get(sp.topicPartition()), sp.topicPartition()))
                 .collect(Collectors.toSet());
```

Comment:
```suggestion
        Set<TopicIdPartition> partitionsWithPendingActions = stopPartitions.stream()
                .filter(sp -> (sp.stopRemoteLogMetadataManager || sp.deleteRemoteLog || sp.deleteLocalLog()) && topicIdByPartitionMap.containsKey(sp.topicPartition()))
                .map(sp -> new TopicIdPartition(topicIdByPartitionMap.get(sp.topicPartition()), sp.topicPartition()))
                .collect(Collectors.toSet());
```

Reply:
I removed `sp.deleteRemoteLog` because we already set `sp.stopRemoteLogMetadataManager` correctly if `sp.deleteRemoteLog` is true. So, only check `(sp.stopRemoteLogMetadataManager || sp.deleteLocalLog)` now.